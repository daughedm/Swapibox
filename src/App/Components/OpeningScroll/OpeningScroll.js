import React from 'react';
import '../OpeningScroll/OpeningScroll.css'

const OpeningScroll = (props) => {

  return (
    <div className="scroll" >
      <div className="fade"></div>
      <section className="star-wars">
        <div className="crawl">
          <div className="title">
            <p>Episode { props.crawl.episode_id }</p>
            <h1>{props.crawl.title}</h1>
          </div>
          <p>{props.crawl.opening_crawl}</p>
        </div>
      </section>
    </div>
  )
}

export default OpeningScroll;