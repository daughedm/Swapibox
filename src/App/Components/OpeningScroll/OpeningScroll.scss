$sw-yellow: rgb(246, 231, 48);

body {
  width: 100%;
  height: 100%;
  // overflow: hidden;
}

.fade {
  width: 100%;
  min-height: 80vh;
  top:  315px;
  background-image: linear-gradient(0deg, transparent, black 75%);
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 800px;
  color: $sw-yellow;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 4px;
  line-height: 150%;
  perspective: 400px;
  text-align: justify;
  z-index: -1;
  white-space: initial;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

.crawl>.title {
  font-size: 90%;
  text-align: center;
}

.crawl>.title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: -100px;
    transform: rotateX(15deg) translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}