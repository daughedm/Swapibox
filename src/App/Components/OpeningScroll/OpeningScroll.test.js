import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import OpeningScroll from './OpeningScroll';

describe('Main', () => {
  let wrapper;

  it('should match snapshot', () => {
    const crawl = {
      "episode_id": "Luke Skywalker",
      "opening_crawl": "Alderaan",
      "title": "Human"
    };

    wrapper = shallow(<OpeningScroll
      crawl={crawl} />);

    expect(wrapper).toMatchSnapshot();
  });
});
