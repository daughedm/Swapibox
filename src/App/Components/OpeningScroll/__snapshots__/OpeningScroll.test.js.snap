// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningScroll
    crawl={
        Object {
            "episode_id": "Luke Skywalker",
            "opening_crawl": "Alderaan",
            "title": "Human",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="fade"
/>,
        <section
          className="star-wars"
>
          <div
                    className="crawl"
          >
                    <div
                              className="title"
                    >
                              <p>
                                        Episode 
                                        Luke Skywalker
                              </p>
                              <h1>
                                        Human
                              </h1>
                    </div>
                    <p>
                              Alderaan
                    </p>
          </div>
</section>,
        <audio
          autoPlay="true"
          src="star-wars-theme-song.mp3"
/>,
      ],
      "className": "scroll",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
>
            <div
                        className="title"
            >
                        <p>
                                    Episode 
                                    Luke Skywalker
                        </p>
                        <h1>
                                    Human
                        </h1>
            </div>
            <p>
                        Alderaan
            </p>
</div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
>
                <p>
                                Episode 
                                Luke Skywalker
                </p>
                <h1>
                                Human
                </h1>
</div>,
              <p>
                Alderaan
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Episode 
                    Luke Skywalker
</p>,
                  <h1>
                    Human
</h1>,
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Episode ",
                      "Luke Skywalker",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Episode ",
                    "Luke Skywalker",
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Human",
                  },
                  "ref": null,
                  "rendered": "Human",
                  "type": "h1",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Alderaan",
              },
              "ref": null,
              "rendered": "Alderaan",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoPlay": "true",
          "src": "star-wars-theme-song.mp3",
        },
        "ref": null,
        "rendered": null,
        "type": "audio",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="fade"
/>,
          <section
            className="star-wars"
>
            <div
                        className="crawl"
            >
                        <div
                                    className="title"
                        >
                                    <p>
                                                Episode 
                                                Luke Skywalker
                                    </p>
                                    <h1>
                                                Human
                                    </h1>
                        </div>
                        <p>
                                    Alderaan
                        </p>
            </div>
</section>,
          <audio
            autoPlay="true"
            src="star-wars-theme-song.mp3"
/>,
        ],
        "className": "scroll",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="crawl"
>
              <div
                            className="title"
              >
                            <p>
                                          Episode 
                                          Luke Skywalker
                            </p>
                            <h1>
                                          Human
                            </h1>
              </div>
              <p>
                            Alderaan
              </p>
</div>,
            "className": "star-wars",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title"
>
                  <p>
                                    Episode 
                                    Luke Skywalker
                  </p>
                  <h1>
                                    Human
                  </h1>
</div>,
                <p>
                  Alderaan
</p>,
              ],
              "className": "crawl",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p>
                      Episode 
                      Luke Skywalker
</p>,
                    <h1>
                      Human
</h1>,
                  ],
                  "className": "title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Episode ",
                        "Luke Skywalker",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "Episode ",
                      "Luke Skywalker",
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Human",
                    },
                    "ref": null,
                    "rendered": "Human",
                    "type": "h1",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alderaan",
                },
                "ref": null,
                "rendered": "Alderaan",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoPlay": "true",
            "src": "star-wars-theme-song.mp3",
          },
          "ref": null,
          "rendered": null,
          "type": "audio",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
